### PR TITLE
perf: reduce filesystem calls

### LINKED ISSUE #157: (perf) Reduce filesystem calls
### Modification Proposal

Reduce filesystem calls, by linting source code provided by webpack, instead of reading modules from filesystem again.

Webpack has an undocumented feature exposing module's original source code via a public method: https://github.com/webpack/webpack/issues/4526

ESLint has a `lintText` method to lint source code directly instead of providing glob patterns (which are expensive to expand) to `lintFiles`: https://eslint.org/docs/developer-guide/nodejs-api#-eslintlinttextcode-options

There may be some other possible optimizations to avoid filesystem calls, e.g. here: (but not sure of the total impact to perf)

https://github.com/webpack-contrib/eslint-webpack-plugin/blob/08af5d683394ec0156aaa61d4c204d4181e76fb0/src/utils.js#L73-L76

### Expected Behavior / Situation

This plugin should (ideally) not need to read from the filesystem at all.
This plugin should be (ideally) very fast!

### Actual Behavior / Situation

This plugin uses many synchronous filesystem calls, directly and indirectly (i.e. via ESLint's `lintFiles` method), and (seem to be) all synchronous. 

This slows down the process quite a bit, especially in certain conditions.

For example, we are developing our application inside a Docker container with source code in a mount to host filesystem. This makes filesystem access quite a bit slower when using WSL2 (which now seems to be standard). Production build executes in about **2 minutes** without this plugin vs about **15 minutes** with this plugin. Dev server (doesn't transpile code with Babel) starts up in about **35 seconds** without this plugin vs about **13 minutes** without this plugin.

### Please paste the results of `npx webpack-cli info` here, and mention other relevant information

Using [Quasar Framework App CLI](https://www.npmjs.com/package/@quasar/app-webpack), not webpack-cli. 
Relevant information is mentioned above. Please let me know if anyone wants any other details.

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
Use the contents of the file using `module.originalSource` or `module._source` reducing calls to files.

Now we use the function the [`lintText`](https://eslint.org/docs/latest/integrate/nodejs-api#-eslintlinttextcode-options) instead of [`lintFiles`](https://eslint.org/docs/latest/integrate/nodejs-api#-eslintlintfilespatterns)

*Need to add the performance improvements

This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [x] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Resolve #157 

### Breaking Changes

No

### Additional Info
No

### CODE REVIEW

**Comment by alexander-akait** [comment]
@ricardogobbosouza Friendly ping, do you need help here?

**Comment by ricardogobbosouza** [comment]
I need to create a large repository to test performance

File: src/worker.js [hunk 1]
```diff
@@ -29,13 +31,28 @@ function setup({ eslintPath, eslintOptions = {} }) {
...
  */
 async function lintFiles(files) {
-  const result = await eslint.lintFiles(files);
-  // if enabled, use eslint autofixing where possible
+  /** @type {LintResult[]} */
+  const results = [];
+
+  await Promise.all(
+    files.map(async (file) => {
+      const result = await eslint.lintText(file.content, {
```
Comment (onigoetz) [inline src/worker.js:42]: Hi, nice PR ! 

ESLint introduced a new `Linter` class that could be better suited to this.

This class also has the advantage of supporting the new ESLint config format: https://eslint.org/blog/2022/08/new-config-system-part-3/#using-flat-config-with-the-linter-class

**Comment by alexander-akait** [comment]
@ricardogobbosouza Do we need this now?

**Pull request closed without merging**